tooling/ast_fuzzer/src/program/func.rs

```diff
@@ -729,4 +733,90 @@ impl<'a> FunctionContext<'a> {
 
         Ok(expr)
     }
+
+    /// Generate a `loop` loop.
+    fn gen_loop(&mut self, u: &mut Unstructured) -> arbitrary::Result<Expression> {
+        // Declare break index variable visible in the loop body.
+        let idx_type = types::U32;
+        let idx_id = self.next_local_id();
+        let idx_name = format!("idx_{}", make_name(idx_id.0 as usize, false));
+        let idx_ident =
+            expr::ident_inner(VariableId::Local(idx_id), true, idx_name.clone(), idx_type);
+
+        // Add a scope which will hold the index variable.
+        //self.locals.add(idx_id, false, idx_name.clone(), idx_type.clone());
+        self.locals.enter();
+
+        // Decrease budget so we don't nest endlessly.
+        self.decrease_budget(1);
+
+        // Initialize index to 0
+        let outer_block = Expression::Block(vec![Expression::Let(Let {
+            id: idx_id,
+            mutable: true,
+            name: idx_name,
+            expression: Box::new(expr::u32_literal(0)),
+        })]);
+
+        let inner_block = self.gen_block(u, &Type::Unit)?;
+        let inner_block = expr::prepend_block(
+            inner_block,
+            vec![Expression::Semi(Box::new(Expression::If(If {
+                condition: Box::new(expr::binary(
+                    Expression::Ident(idx_ident.clone()),
+                    BinaryOp::Equal,
+                    expr::u32_literal(self.ctx.config.max_loop_size as u32),
+                )),
+                consequence: Box::new(Expression::Block(vec![Expression::Break])),
+                alternative: None,
+                typ: Type::Unit,
+            })))],
+        );
+        let inner_block = expr::extend_block(
+            inner_block,
+            vec![Expression::Assign(Assign {
```

Comment:
This could be in the `else` of the check above, so the two related statements are close together in the AST.